keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -1529,7 +1550,24 @@ kpxc.fillIn = function(combination, onlyPassword, suppressWarnings) {
         }
 
         if (pField) {
-            kpxc.setValueWithChange(pField, kpxc.credentials[combination.loginId].password);
+            let password = kpxc.credentials[combination.loginId].password;
+            if (appendTOTP) {
+                if (kpxc.credentials[commbination.loginId].stringFields && kpxc.credentials[combination.loginId].stringFields.length > 0) {
+                    const stringFields = kpxc.credentials[combination.loginId].stringFields;
```

Comment:
Haven't tried your PR yet, but try to get rid of this duplicate code. It's almost identical to the stuff above. Maybe a separate function (inside the function?) for it?

Reply:
I refactored the code to use a new _'kpxc.getTOTP'_ function. I hope this looks good to you